src/main/java/com/google/api/codegen/transformer/java/JavaModelTypeNameConverter.java

```diff
@@ -227,6 +231,21 @@ public TypeName getTypeNameForTypedResourceName(
     }
   }
 
+  private static String getResourceNamePackage(ProtoFile file, ResourceNameType resourceNameType) {
```

Comment:
My preference would be for the type (resourceNameType) to know what package itself is in (i.e. all of the logic here); that bundles the concept together more coherently. Basically I think the "implicit package" analog in this case is actually inside `ResourceNameType`.